lib/modules/datasource/docker/index.ts

```diff
@@ -558,6 +594,66 @@ export class DockerDatasource extends Datasource {
     return null;
   }
 
+  @cache({
+    namespace: 'datasource-docker-architecture',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      currentDigest: string
+    ) => `${registryHost}:${dockerRepository}@${currentDigest}`,
+    ttlMinutes: 1440 * 28,
+  })
+  public async getImageArchitecture(
+    registryHost: string,
+    dockerRepository: string,
+    currentDigest: string
+  ): Promise<string | null | undefined> {
+    try {
+      const manifestResponse = await this.getManifestResponse(
+        registryHost,
+        dockerRepository,
+        currentDigest,
+        'head'
+      );
+
+      if (
+        manifestResponse?.headers['content-type'] === MediaType.manifestV2 ||
+        manifestResponse?.headers['content-type'] === MediaType.ociManifestV1
+      ) {
+        const configDigest = await this.getConfigDigest(
+          registryHost,
+          dockerRepository,
+          currentDigest
+        );
+        if (!configDigest) {
+          return undefined;
```

Comment:
IIRC that found its way in there because of https://github.com/renovatebot/renovate/pull/16554#discussion_r924108859. 

Thinking about it, I agree that null would be a better choice here:

- If an error occurs when fetching the config manifest (e.g. invalid `schemaVersion`), it would likely happen again.
- Returning null (and having it cached then) is analogous to what is done in `getLabels()`, in case of error: https://github.com/renovatebot/renovate/blob/f01d5eda278fc18289194681ac1ad5e328c79ede/lib/modules/datasource/docker/index.ts#L677-L684